event-sourcing/event-store-cassandra/src/test/java/org/apache/james/eventsourcing/eventstore/cassandra/CassandraEventStoreExtension.java

```diff
@@ -37,13 +38,19 @@
 import com.google.common.collect.ImmutableSet;
 
 public class CassandraEventStoreExtension implements BeforeAllCallback, AfterAllCallback, BeforeEachCallback, AfterEachCallback, ParameterResolver {
-    @SuppressWarnings("rawtypes")
-    private final Set<EventDTOModule> modules;
-    private CassandraClusterExtension cassandra;
+    private final CassandraClusterExtension cassandra;
+    private final Supplier<JsonEventSerializer> serializerSupplier;
     private EventStoreDao eventStoreDao;
 
     public CassandraEventStoreExtension(@SuppressWarnings("rawtypes") EventDTOModule... modules) {
-        this.modules = Arrays.stream(modules).collect(ImmutableSet.toImmutableSet());
+        @SuppressWarnings("rawtypes")
+        Set<EventDTOModule> savedModules = Arrays.stream(modules).collect(ImmutableSet.toImmutableSet());
+        serializerSupplier = () -> new JsonGenericEventSerializer(savedModules);
+        this.cassandra = new CassandraClusterExtension(CassandraEventStoreModule.MODULE);
+    }
+
+    public CassandraEventStoreExtension(Supplier<JsonEventSerializer> serializerSupplier) {
```

Comment:
you should not have two constructors doing field initialization. Only one and others must call it.

Reply:
discarded